Modules/_elementtree.c

```diff
@@ -3894,6 +3894,34 @@ _elementtree_XMLParser_close_impl(XMLParserObject *self)
     }
 }
 
+/*[clinic input]
+_elementtree.XMLParser.flush
+
+[clinic start generated code]*/
+
+static PyObject *
+_elementtree_XMLParser_flush_impl(XMLParserObject *self)
+/*[clinic end generated code: output=42fdb8795ca24509 input=effbecdb28715949]*/
+{
+    if (!_check_xmlparser(self)) {
+        return NULL;
+    }
+
+    elementtreestate *st = self->state;
+
+    if (EXPAT(st, SetReparseDeferralEnabled) == NULL) {
```

Comment:
It should check the result of `GetReparseDeferralEnabled()` as well.

Reply:
@serhiy-storchaka I thought so too, until I realized that the `self->parser` in etree is fully controlled by etree and not exposed to the outside; as a result it's impossible that both (1) `EXPAT(st, SetReparseDeferralEnabled)` is available and (2) reparse deferrall is disabled together. Hence I dropped the code calling `GetReparseDeferralEnabled` in etree.

Reply:
It is for good! Thank you for adding the comment.